code/modules/hydroponics/hydroponics.dm

```diff
@@ -75,6 +75,7 @@
 	plant_hud_set_water()
 
 /obj/machinery/hydroponics/Destroy()
+	remove_from_all_data_huds()
```

Comment:
I'm not sure if this is how huds work, but this seems to function correctly, allows hydro trays to GC, and didn't generate any runtimes or oddities that I could spot. @Crazylemon64 @Aurorablade